src/main/java/ca/on/oicr/pineryreports/reports/impl/LibrariesBillingReport.java

```diff
@@ -0,0 +1,301 @@
+import static ca.on.oicr.pineryreports.util.SampleUtils.byCreatedBetween;
+import static ca.on.oicr.pineryreports.util.SampleUtils.getAttribute;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Option;
+import org.apache.commons.cli.ParseException;
+
+import com.google.common.collect.Sets;
+
+import ca.on.oicr.pinery.client.HttpResponseException;
+import ca.on.oicr.pinery.client.PineryClient;
+import ca.on.oicr.pinery.client.SampleClient.SamplesFilter;
+import ca.on.oicr.pineryreports.data.ColumnDefinition;
+import ca.on.oicr.pineryreports.reports.TableReport;
+import ca.on.oicr.pineryreports.util.CommonOptions;
+import ca.on.oicr.ws.dto.SampleDto;
+
+public class LibrariesBillingReport extends TableReport {
+
+	private static class SummaryObject {
+		private final String project;
+		private final String kit;
+		private Integer libsCount = 0;
+		
+		public SummaryObject(String project, String kit) {
+		  this.project = project;
+		  this.kit = kit;
+		  this.libsCount = 1;
+		}
+		
+		public String getProject() {
+		  return project;
+		}
+		
+		public String getKit() {
+		  return kit;
+		}
+		
+		public Integer getLibrariesCount() {
+		  return libsCount;
+		}
+		
+		public void addOneLibrary() {
+		  libsCount += 1;
+		}
+		
+		private static final Comparator<SummaryObject> summaryComparator = new Comparator<SummaryObject>() {
+      @Override
+      public int compare(SummaryObject o1, SummaryObject o2) {
+        if (o1.getProject().equals(o2.getProject())) {
+          return o2.getKit().compareTo(o2.getKit());
+        } else {
+          // sort on this primarily
+          return o1.getProject().compareTo(o2.getProject());
+        }
+      }
+    };
+  }
+
+  private static class DetailedObject {
+    private final String creationDate;
+    private final String kit;
+    private final String libraryAlias;
+    private final String libraryDesignCode;
+    private final String project;
+    
+    public DetailedObject(String project, String kit, String creationDate, String libraryAlias, String libraryDesignCode) {
+      this.project = project;
+      this.kit = kit;
+      this.creationDate = timeStringToYyyyMmDd(creationDate);
+      this.libraryAlias = libraryAlias;
+      this.libraryDesignCode = libraryDesignCode;
+    }
+    
+    public String getCreationDate() {
+      return creationDate;
+    }
+    
+    public String getKit() {
+      return kit;
+    }
+    
+    public String getLibrary() {
+      return libraryAlias;
+    }
+    
+    public String getLibraryDesignCode() {
+      return libraryDesignCode;
+    }
+    
+    public String getProject() {
+      return project;
+    }
+    
+    private static final Comparator<DetailedObject> detailedComparator = new Comparator<DetailedObject>() {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Make this anonymous inner class a lambda [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1604)